.github/workflows/build.yml

```diff
@@ -16,3 +16,5 @@ jobs:
       - uses: jdx/mise-action@146a28175021df8ca24f8ee1828cc2a60f980bd5 # v3.5.1
       - name: Run tests
         run: mise run check
+        env:
```

Comment:
I wonder if this should be a separate workflow.
It won't make the overall build much faster, but provide faster fail when unit tests fail.